File: rails-api/app/models/chat_room.rb
class ChatRoom < ApplicationRecord
  # validations
  validates_presence_of :title

  belongs_to :sender, foreign_key: :sender_id, class_name: 'User'
  belongs_to :recipient, foreign_key: :recipient_id, class_name: 'User'

  has_many :messages, dependent: :destroy
end

File: client-side/src/app/models/chat-room.ts
export class ChatRoom {
  constructor(
    public id = 0,
    public title = '',
    public created_id = 1,
    public sender_id = 1,
    public recipient_id = 1) { }
  clone() { return new ChatRoom(this.id, this.title, this.created_id, this.sender_id, this.recipient_id); }
}
